onnxruntime/core/providers/mkldnn/mkldnn_execution_provider.h

```diff
@@ -37,6 +44,38 @@ class MKLDNNExecutionProvider : public IExecutionProvider {
   }
 
   virtual std::shared_ptr<KernelRegistry> GetKernelRegistry() const override;
+
+  std::shared_ptr<mkldnn::memory> GetWeightsMemory(const std::string& weight_key) {
+    std::lock_guard<std::mutex> lock(mutex_);
+    auto iter = weights_mem_map_.find(weight_key);
+    if (iter != weights_mem_map_.end())
+      return iter->second;
+    return nullptr;
+  }
+
+  void SetWeightsMemory(const std::string& weight_key,
+                        const std::shared_ptr<mkldnn::memory>& filter_dst_mem) {
+    std::lock_guard<std::mutex> lock(mutex_);
+    weights_mem_map_.insert(std::make_pair(weight_key, filter_dst_mem));
+  }
+
+  std::mutex& GetMutex() {
+    return conv_mutex_;
+  }
+
+  void SaveAllocatedMemory(IAllocatorUniquePtr<void> buffer) {
```

Comment:
prefer a more descriptive name. maybe SaveWeightsMemoryBuffer

Reply:
Done. SaveWeightsMemoryBuffer and GetWeightsMemoryBuffer